cluster-autoscaler/cloudprovider/cloud_provider.go

```diff
@@ -56,6 +56,9 @@ type CloudProvider interface {
 	// GetResourceLimiter returns struct containing limits (max, min) for resources (cores, memory etc.).
 	GetResourceLimiter() (*ResourceLimiter, error)
 
+	// GetInstanceID gets the instance ID for the specified node.
+	GetInstanceID(node *apiv1.Node) string
```

Comment:
Are we 100% percent sure that this is the right approach? Honestly, I have doubts that a (hopefully) temporary bugfix/workaround for minor issues in NodeController/Azure node registration process should involve an api change for all cloud providers (including ones that are not in this repo). Do we see any other applications of this method other "toLower" in Azure?

Reply:
I don't expect to change all the cloud providers, but since the providerID is handled in clusterstate and it is case sensitive, the changes are still required. (there are also other ways, e.g. only this for Azure, but that's a little hacky).

In other applications, we are also keeping the resourceID of Azure VM, but when the resourceID is used for comparison, it is converted to lower or upper cases.

Reply:
I'd like to understand if there is a plan to fix it in Azure (ie. make ProviderID match the name of VM)? Basically is this a temporary workaround? Are we planning to remove it later on? If so I'd like a comment stating that.

Reply:
@MaciekPytel This is actually for fixing the issues with old releases, as providerID couldn't be changed after node initialized. 

For new releases, we should fix the providerID inconsistent issues in the cloud provider, probably cast to lower cases instead of using VM name.

Reply:
Are you talking about master or node versions? How long will we support these old releases? What should w do in, let's say half a year from now - remove this method?

Reply:
I mean cherry pick to old stable releases (for each kubernetes release). Those workarounds should be there and never get deleted. But for master branch, we may delete it after fixing the issues in cloud provider (e.g. in v1.14).

Reply:
Ok, makes sense.